generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/extensions/kClassExtensions.kt

```diff
@@ -84,16 +84,16 @@ internal fun KClass<*>.isListType(isDirective: Boolean = false): Boolean = this.
 
 @Throws(CouldNotGetNameOfKClassException::class)
 internal fun KClass<*>.getSimpleName(isInputClass: Boolean = false): String {
-    val name = this.getGraphQLName()
-        ?: this.simpleName
-        ?: throw CouldNotGetNameOfKClassException(this)
+
+    val name = this.getGraphQLName() ?: this.simpleName ?: throw CouldNotGetNameOfKClassException(this)
+
+    this.getGraphQLNameTarget().takeIf { it == GraphQLNameTarget.INPUT }?.let { return name }
```

Comment:
the intent behind new `GraphQLNameTarget` is to better control how this name should be applied

```
when target {
  GraphQLNameTarget.INPUT -> if (isInputClass) this.getGraphQLName else <some exception that annotation is applied on output type and suggest to change it to target both>
  GraphQLNameTarget.OUTPUT -> if (!isInputClass) this.getGraphQName else <some exception that we cannot apply it on input object and suggest to change it to target both>
  GraphQLNameTarget.BOTH -> if (isInputClass) this.getGraphQLname + input suffix else this.getGraphQLName
```

or something along those lines

Reply:
Thanks for your input, I have made the suggested changes. Could you please revalidate? Also if you could suggest on the docs If I need to include something else or some more information around the same?